android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -248,6 +248,15 @@ public void processProperties(KrollDict d)
 				tv.setImeOptions(EditorInfo.IME_FLAG_NO_FULLSCREEN);
 			}
 		}
+
+		if (d.containsKey(TiC.PROPERTY_LINES)) {
+			if (!field) tv.setLines(TiConvert.toInt(d, TiC.PROPERTY_LINES));
+		}
+
+		if (d.containsKey(TiC.PROPERTY_MAX_LINES)) {
+			if (!field) tv.setMaxLines(TiConvert.toInt(d, TiC.PROPERTY_MAX_LINES));
```

Comment:
From looking at Google's `Textview` Java code, they don't expect `setMaxLines()` to be set to a value <= 0. Google defaults this to `Integer.MAX_VALUE` by default. We should do the same if the property is set to <= 0. This is to avoid possible layout issues.
https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/android/widget/TextView.java#L658

I believe we might have a similar issue with Android's `setLines()` method since it set private member variable `mMaximum` (used as max lines) to the given `lines` value.
https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/android/widget/TextView.java#L4616

So, if the `lines` property is <= 0, then we should call `setMinLines(0)` instead.

I think the best thing to do is to apply `lines` and `maxLines` property changes via a method like how I'm doing it in the code below. You don't need the `setSingleLine()` call, but you do need to call `setMaxLines()` after calling `setLines()` since Google resets the maximum afterwards.
https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java#L706

Reply:
ah nice, I'll copy the parts over to the UIText!